controllers/common/ansible.go

```diff
@@ -44,10 +45,28 @@ func CreateAnsibleJob(policyAutomation *policyv1beta1.PolicyAutomation,
 		ansibleJob.Object["spec"].(map[string]interface{})["extra_vars"] = mapExtraVars
 	}
 
-	if targetClusters != nil {
-		// nolint: forcetypeassert
+	if violationContext.TargetClusters != nil {
```

Comment:
If it's nil, could you please use an empty map instead? This makes the extra vars "API" more consistent.

Reply:
```
  extra_vars:
    hub_cluster: ""
    namespace: ""
    policy_name: ""
    policy_set: null
    policy_violation_context: null
    sn_priority: 1
    sn_severity: 1
    target_clusters: null
```

Reply:
@ChunxiAlexLuo is this actually the case? I was expecting `policy_violation_context` to be `{}` and `target_clusters` to be `[]`.

Reply:
@mprahl Yes, this is the actual data for `manual` mode. When `targetList` is empty, getViolationContext() will not be called and CreateAnsibleJob() only be called under `manual` mode. Since we need to keep the `mapExtraVars` value from `policyAutomation.Spec.Automation.ExtraVars`, I insert the value from `violationContext` into `mapExtraVars`.

```
		log.Info("Creating an Ansible job", "mode", "manual")

		err = common.CreateAnsibleJob(
			policyAutomation,
			r.DynamicClient,
			"manual",
			policyv1beta1.ViolationContext{},
		)
```

Reply:
Okay, so the `mapExtraVars[fieldName] = value` line must just be assigning `nil` instead of `[]`/`{}` if the slice/map is empty. Is there a way to guard against this?

Reply:
@mprahl Tested some ways but don't find an easy way,  except the way that set these fields to `[]/{}` later when checking them are `nil`, which looks ugly and inflexible when adding new fields.